File: VB/DXWebApplication_ReportDesigner/DAL/ReportEntity.vb
Imports DevExpress.Xpo

Namespace DXWebApplication_ReportDesigner.DAL
	<DeferredDeletion(False)>
	Public Class ReportEntity
		Inherits XPObject

'INSTANT VB NOTE: The variable name was renamed since Visual Basic does not allow variables and other class members to have the same name:
		Private name_Renamed As String
'INSTANT VB NOTE: The variable layout was renamed since Visual Basic does not allow variables and other class members to have the same name:
		Private layout_Renamed() As Byte

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", name_Renamed, value)
			End Set
		End Property


		Public Property Layout() As Byte()
			Get
				Return layout_Renamed
			End Get
			Set(ByVal value As Byte())
				SetPropertyValue("Layout", layout_Renamed, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/DXWebApplication_ReportDesigner/DAL/ReportEntity.cs
using DevExpress.Xpo;

namespace DXWebApplication_ReportDesigner.DAL {
    [DeferredDeletion(false)]
    public class ReportEntity : XPObject {
        string name;
        byte[] layout;

        public ReportEntity(Session session)
            : base(session) {
        }

        public string Name {
            get { return name; }
            set { SetPropertyValue("Name", ref name, value); }
        }


        public byte[] Layout {
            get { return layout; }
            set { SetPropertyValue("Layout", ref layout, value); }
        }
    }
}